src/term/check/unbound_pats.rs

```diff
@@ -39,10 +40,12 @@ impl Pattern {
           }
           check.extend(args.iter());
         }
-        Pattern::Var(_) => (),
-        Pattern::Num(_) => (),
-        Pattern::Tup(_, _) => (),
+        Pattern::Tup(fst, snd) => {
+          unbounds.extend(fst.unbound_pats(is_ctr));
+          unbounds.extend(snd.unbound_pats(is_ctr));
+        }
         Pattern::List(..) => unreachable!(),
```

Comment:
Can you implement this for Pattern::List?
I feel like this pass is so basic that at some point in the future we might change the order of things.